WcaOnRails/app/models/competition.rb

```diff
@@ -458,11 +465,82 @@ def in_progress?
   end
 
   def is_over?
-    end_date < Date.today
+    !!end_date && end_date < Date.today
+  end
+
+  def results_updated_at
+    Option.from_nilable(results.order('updated_at desc').pluck(:updated_at).first)
+  end
+
+  def result_cache_key(view)
+    [id, view, results_updated_at.map(&:iso8601).unwrap_or("")]
```

Comment:
Interesting. That sounds like it could be a useful idiom once you're used to it. Thanks for the details.

What about avoiding this problem entirely? In this particular case, if there are no Results for this competition, we could use the competition's updated_at as a cache key. Oh wait, Competitions don't have an updated_at field... What about adding an `updated_at` field, or using a Competition's start date?